app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -3,26 +3,35 @@ module Broadcasts
   module WelcomeNotification
     class Generator
       def initialize(receiver_id)
-        @receiver_id = receiver_id
+        @user = User.find(receiver_id)
       end
 
       def self.call(*args)
         new(*args).call
       end
 
       def call
-        # This method should find the user based on the `receiver_id`.
-        # It should then determine the appropriate Broadcast for a user,
-        # based on the `created_at` and the different conditions for sending a notification.
-        # `welcome_broadcast = ...`
+        return if commented_on_welcome_thread? || received_notification?
 
-        # Once it has the appropriate Broadcast to be sent, it should send a notification for it:
-        # `Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)`
+        Notification.send_welcome_notification(user.id, welcome_broadcast.id)
+      end
+
+      def received_notification?
+        Notification.exists?(notifiable: welcome_broadcast, user: user)
+      end
+
+      def commented_on_welcome_thread?
+        welcome_threads = Article.where("title LIKE 'Welcome Thread - %'").order(created_at: :desc).first
```

Comment:
```suggestion
        welcome_thread = Article.where("title LIKE 'Welcome Thread - %'").order(created_at: :desc).first
```

This should just be _one_ thread, not multiple, right? I also think we should use the logic [from here](url) as @maestromac suggested.

Reply:
Correct! And I must have misread your comment to Mac, because I thought you were postponing the logic from Mac's suggestion until you picked up the `welcome_thread` issue. 🙃 I'll make the change!

Reply:
No worries, that thread got kind of long! 😂 I think I landed on this: we should copy-pasta the logic into here, and then we can refactor it at a later time into a scope on the `Article` class :)

Reply:
Update: I've successfully copy-pasta'd and made this change! 😄 😂